File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/core/date/DateConverter.java
package coolpharaoh.tee.speicher.tea.timer.core.date;

import androidx.room.TypeConverter;

import java.util.Date;

public class DateConverter {

    private DateConverter() {
    }

    @TypeConverter
    public static Date toDate(final Long dateLong) {
        return dateLong == null ? null : new Date(dateLong);
    }

    @TypeConverter
    public static Long fromDate(final Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/core/date/DateConverter.kt
package coolpharaoh.tee.speicher.tea.timer.core.date

import androidx.room.TypeConverter
import java.util.Date

object DateConverter {
    @JvmStatic
    @TypeConverter
    fun toDate(dateLong: Long?): Date? {
        return if (dateLong == null) null else Date(dateLong)
    }

    @JvmStatic
    @TypeConverter
    fun fromDate(date: Date?): Long? {
        return date?.time
    }
}